syntax = "proto3";
option cc_enable_arenas = true;

package Ydb.Nbs;

option java_package = "com.yandex.ydb.nbs.proto";
option java_outer_classname = "NbsProtos";
option java_multiple_files = true;

import "ydb/public/api/protos/ydb_operation.proto";


////////////////////////////////////////////////////////////////////////////////
// Partition create request/response.

message CreatePartitionRequest {
    Ydb.Operations.OperationParams operation_params = 1;

    // Storage pool name to use.
    string StoragePoolName = 2;

    // Minimum addressable block size (smallest unit of I/O operations).
    uint32 BlockSize = 3;

    // Maximum number of blocks stored in partition.
    uint64 BlocksCount = 4;
}

message CreatePartitionResponse {
    Ydb.Operations.Operation operation = 1;
}

message CreatePartitionResult {
    string TabletId = 1;
}

////////////////////////////////////////////////////////////////////////////////
// Partition delete request/response.

message DeletePartitionRequest {
    Ydb.Operations.OperationParams operation_params = 1;
    // Partition tablet id to delete.
    string TabletId = 2;
}

message DeletePartitionResponse {
    Ydb.Operations.Operation operation = 1;
}

message DeletePartitionResult {
    // Deleted partition tablet id.
    string TabletId = 1;
}

////////////////////////////////////////////////////////////////////////////////
// Partition list request/response.

message ListPartitionsRequest {
    Ydb.Operations.OperationParams operation_params = 1;
}

message ListPartitionsResponse {
    Ydb.Operations.Operation operation = 1;
}

message ListPartitionsResult {
    repeated string TabletId = 1;
}
